component/src/main/java/org/wso2/extension/siddhi/store/influxdb/InfluxDBStore.java

```diff
@@ -0,0 +1,565 @@
+     */
+    @Override
+    protected void updateOrAdd(CompiledCondition compiledCondition, List<Map<String, Object>> list,
+                               Map<String, CompiledExpression> map, List<Map<String, Object>> listToAdd,
+                               List<Object[]> listToUpdate) {
+
+        try {
+
+            this.writePoints(listToUpdate);
+
+        } catch (InfluxDBException e) {
+            throw new InfluxDBTableException("Error in updating or inserting records to table '" + this.tableName
+                    + "': " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Compile the matching condition
+     *
+     * @param expressionBuilder that helps visiting the conditions in order to compile the condition
+     * @return compiled condition that can be used for matching events in find, contains, delete, update and
+     * updateOrAdd
+     */
+    @Override
+    protected CompiledCondition compileCondition(ExpressionBuilder expressionBuilder) {
+
+        InfluxDBConditionVisitor visitor = new InfluxDBConditionVisitor();
+        expressionBuilder.build(visitor);
+        return new InfluxDBCompiledCondition(visitor.returnCondition(), visitor.getParameters(),
+                visitor.getParametersConstant());
+
+    }
+
+    /**
+     * Compile the matching condition
+     *
+     * @param expressionBuilder that helps visiting the conditions in order to compile the condition
+     * @return compiled condition that can be used for matching events in find, contains, delete, update and
+     * updateOrAdd
+     */
+    @Override
+    protected CompiledExpression compileSetAttribute(ExpressionBuilder expressionBuilder) {
+
+        return compileCondition(expressionBuilder);
+    }
+
+    /**
+     * This method will be called before the processing method.
+     * Intention to establish connection to publish event.
+     *
+     * @throws ConnectionUnavailableException if end point is unavailable the ConnectionUnavailableException thrown
+     *                                        such that the  system will take care retrying for connection
+     */
+    @Override
+    protected void connect() throws ConnectionUnavailableException {
+
+        try {
+            influxdb = InfluxDBFactory.connect(url, username, password);
+            if (!this.checkDatabaseExists(database)) {
+
+                throw new ConnectionUnavailableException("database " + database + " does not exist");
+            }
+            influxdb.setDatabase(database);
+        } catch (InfluxDBIOException e) {
+            throw new ConnectionUnavailableException("failed to initialize influxDB store " + e.getMessage(), e);
+        } catch (InfluxDBException.AuthorizationFailedException e) {
+            throw new ConnectionUnavailableException("wrong url or username " + e.getMessage(), e);
+        }
+
+    }
+
+    @Override
+    protected void disconnect() {
+
```

Comment:
Disconnect and destroy are not implemented

Reply:
https://github.com/wso2-extensions/siddhi-store-influxdb/pull/1/commits/520b093086a08e2d8d9df4297dcd71914e15813b